test/cpp/end2end/xds/xds_end2end_test.cc

```diff
@@ -1951,7 +1945,7 @@ class XdsEnd2endTest : public ::testing::TestWithParam<TestType> {
 
 class BasicTest : public XdsEnd2endTest {
  public:
-  BasicTest() : XdsEnd2endTest(4, 1) {}
+  BasicTest() : XdsEnd2endTest(4, 1) { StartAllBackends(); }
```

Comment:
Looks like you missed one other spot where this change is needed:

https://github.com/grpc/grpc/blob/b1e6b1f8ccff2dabdaaee6cff1d6a2c14cb09cf1/test/cpp/end2end/xds/xds_end2end_test.cc#L2584

Here's an alternative that might be less invasive:

- Rename `XdsEnd2endTest` to `XdsEnd2endTestBase`.
- Create a subclass called `XdsEnd2endTest` whose ctor calls `StartAllBackends()`.
- Change `XdsEnabledServerTest` and `XdsServerSecurityTest` to inherit from `XdsEnd2endTestBase` instead of `XdsEnd2endTest`.

Reply:
I think as you suggested each test suite should make its own decision on what backends to start. Explicitly seeing the `StartAllBackends()` method would make it easier to iteratively change ALL tests such that they only start the backends that they need.